### PR TITLE
Changelog and version bump for 5.1.2

### PR DESCRIPTION
**Pull request by vkbo**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+* Fixed a little bug in aperture which becomes evident if `tlen` starts to deviate from `dcum(iu)`. All the calculations in aperture (especially for the backtracking algorithm) are based on `dcum`, but if the code has to deal with the beginning/end of the ring, it was using `tlen`, whereas, to be consistent, it should use `dcum(iu)`. PR #755
+
+**Code Improvements and Changes**
+
+* The two modules handling file units have been merged to a single module. Each previous module was handling dynamic allocation of file unit and wrapping the open statement for compiler options separately. The new module now also writes a log file listing all files opened and closed through the module. PR #747
+* The parsing of the aperture limitations file in the `LIMI` block has been updated and made more consistent with how other secondary input files are handled elsewhere in SixTrack. This simplifies the code. PR #754
+
```
Comment (amereghe) [inline CHANGELOG.md]: Actually, the changes in the way the `LIMI` block is parsed also remove also another bug, i.e. any keyword-pair in the block appearing after the `LOAD` statement was ignored.

**Comment by amereghe** [comment]
Excellent! Thanks a lot, Veronica!

**Comment by kyrsjo** [comment]
Add a comment about #665 :
 * No-longer used particle array size variables in the collimation code were cleaned up.

**Comment by vkbo** [comment]
Yeah, I haven't added the last three PRs yet.

**Pull request merged**